crates/containerd-shim-wasmtime/Cargo.toml

```diff
@@ -12,26 +12,8 @@ oci-spec = { workspace = true, features = ["runtime"] }
 ttrpc = { workspace = true }
 sha256 = { workspace = true }
 
-# We are not including the `async` feature here:
-# 1. Because we don't even use it
-# 2. Because it pulls in a lot of dependencies that we don't need
-# 3. Because that dependency (wasmtime-fiber) links to native code
-# 4. The wasmedge shim also uses wasmtime-fiber... which means those transative dependencies need to be the same or compilation fails
-wasmtime = { version = "17.0", default-features = false, features = [
-    "cache",
-    "wat",
-    "profiling",
-    "parallel-compilation",
-    "cranelift",
-    "pooling-allocator",
-    "demangle",
-    "addr2line",
-    "coredump",
-    "debug-builtins",
-    'component-model',
-]}
-wasmtime-wasi = { version = "17.0", features = ["exit"] }
-wasi-common = "17.0"
+wasmtime = { version = "20.0.2", features = ["async"] }
```

Comment:
[Wasmtime 21 was released](https://github.com/bytecodealliance/wasmtime/blob/main/RELEASES.md)
I saw that by updating the version the repository compiles and the tests pass. If you prefer, I open a new PR or rename this one with the transition directly to wasmtime 21

Reply:
Sure, Would you mind doing the bump to Wasmtime 21 as a separate PR?  I am still not sure the implications of turning on the async support and a few of the other options enabled here.  I need a little more time to understand what turning these features means.  Maybe other reviewers have some thoughts too

Reply:
> Sure, Would you mind doing the bump to Wasmtime 21 as a separate PR? I am still not sure the implications of turning on the async support and a few of the other options enabled here. I need a little more time to understand what turning these features means. Maybe other reviewers have some thoughts too

For the transition to wasmtime 21 in my opinion we can wait as long as, we simply need to update the dependencies of Cargo.toml.

I'm currently testing the new version on my k8s.

Please take the time to evaluate the impacts of the proposed changes. As already indicated I attempted to keep the code with async disabled but I had got failed tests so I opted to switch to async mode. This should not impact other shims.

I hope other people can try my modifications on more serious implementations than the my toy one with protohackers.